internal/functions/efgs/redis/redis.go

```diff
@@ -0,0 +1,83 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	redisclient "github.com/go-redis/redis/v8"
+	"os"
+	"sync"
+	"time"
+)
+
+var redisClient *Connection
+var ctx context.Context
+
+type lazyConnection func() *redisclient.Client
+
+//Connection Contains lazy Redis connection
+type Connection struct {
+	inner lazyConnection
+}
+
+func init() {
+	ctx = context.Background()
+
+	connect := func() *redisclient.Client {
+		logger := logging.FromContext(ctx).Named("efgs.redis.connect")
+
+		logger.Debug("Connecting to EFGS Redis")
+
+		addr, ok := os.LookupEnv("EFGS_REDIS_ADDR")
+		if !ok {
+			panic("EFGS_REDIS_ADDR env missing")
+		}
+
+		client := redisclient.NewClient(&redisclient.Options{
+			Addr: addr,
+			DB:   0,
+		})
+
+		_, err := client.Ping(ctx).Result()
+		if err != nil {
+			panic(fmt.Sprintf("Connection to Redis failed:%v", err))
+		}
+
+		logger.Debugf("Connected to EFGS Redis at %v", addr)
+
+		return client
+	}
+
+	var conn *redisclient.Client
+	var once sync.Once
+
+	initInner := func() *redisclient.Client {
+		once.Do(func() {
+			conn = connect()
+		})
+		return conn
+	}
+
+	redisClient = &Connection{
+		inner: initInner,
+	}
+}
+
+//Client Redis client abstraction
+type Client interface {
+	Get(key string) (string, error)
+	Set(key string, value interface{}, ttl time.Duration) error
+}
+
+//RealClient Real Redis client
+type RealClient struct{}
```

Comment:
I think a better and more used name would be `ClientImpl` rather than `RealClient`.